### PR TITLE
Create create-azure-resource.md

### PR DESCRIPTION
**Pull request by phalbert**
# Description

This guide has an example resource in Azure using Terraform. This is executed with a Jenkins pipeline.

## Added docs pages

Please also include the path for the added docs

- Create resource in Azure Cloud with Terraform (`./create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource`)

### CODE REVIEW

**Comment by phalbert** [comment]
Need to use port terraform provider

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+Follow these steps to get started:
+
+1. Create the following as Jenkins Credentials:
+    1. Create the Port Credentials using the `Username with password` kind.
+        1. `PORT_CLIENT_ID` - Port Client ID [learn more](../../../../build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
+        2. `PORT_CLIENT_SECRET` - Port Client Secret [learn more](../../../../build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
+    2. Create the Azure Credentials using the `Azure Service Principal` kind.
+        1. `ARM_CLIENT_ID` - Azure Client ID (APP ID) of the application.
+        2. `ARM_CLIENT_SECRET` - Azure Client Secret (Password) of the application.
+        3. `ARM_SUBSCRIPTION_ID` - Azure Subscription ID.
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:27]: Does Azure have some documentation guide showing where to get this information? If there, let's add a link here, same for `ARM_TENANT_ID`

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+        4. `ARM_TENANT_ID` - The Azure Tenant ID.
+    3. `WEBHOOK_TOKEN` - The webhook token so that the job can only be triggered if that token is supplied.
+
+:::tip
+Follow this [article](https://learn.microsoft.com/en-us/cli/azure/azure-cli-sp-tutorial-1?tabs=bash) to create a service principal in order to get the Azure credentials.
+:::
+
+2. Create a Port blueprint with the following properties:
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:35]: Add a tooltip on the word blueprint, similar to how it's done in the quickstart: https://docs.getport.io/quickstart#4-set-standards-using-scorecards

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+    "mirrorProperties": {},
+    "calculationProperties": {},
+    "relations": {}
+}
+```
+
+  </details>
+
+
+3. Create Port Action from [here](https://app.getport.io/self-serve) using the following JSON definition:
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:84]:
```suggestion
3. Create a Port Action in the [self-service hub](https://app.getport.io/self-serve) using the following JSON definition:
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+
+:::note
+Please make sure to modify JENKINS_HOST and TOKEN placeholders to match your environment.
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:87]:
```suggestion
Make sure to replace the placeholders for `JENKINS_URL` and `JOB_TOKEN`.
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+:::
+
+  <details>
+  <summary>Port Action</summary>
+
+```json showLineNumbers
+{
+    "identifier": "create_azure_storage",
+    "title": "Create Azure Storage",
+    "icon": "S3",
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:97]:
```suggestion
    "icon": "Azure",
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+                        def payloadJson = jsonSlurper.parseText(result.trim())
+
+                        // Access the desired data from the payload
+                        PORT_ACCESS_TOKEN = payloadJson.accessToken
+                    }
+                }
+            }
+        }
+        
+        stage('Terraform Azure') {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:328]: Are both this and the `Terraform Port` steps needed? it seems like they are doing pretty much the same thing, only one doesn't echo in the middle.

Can we remove one of them?

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 7]
```diff
@@ -0,0 +1,460 @@
...
+        "method": "POST"
+    },
+    "trigger": "CREATE",
+    "requiredApproval": false
+}
+```
+
+  </details>
+
+4. Create a Terraform templates in your GitHub repository:
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:134]:
```suggestion
4. Create a Terraform template in your GitHub repository:
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 8]
```diff
@@ -0,0 +1,460 @@
...
+
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:135]: This section requires a bit more of an explanation - outline ahead of the time what files are we going to create, why are we creating each one, and where in the repository are they expected to be

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 9]
```diff
@@ -0,0 +1,460 @@
...
+This example demonstrates how to deploy a storage account in Azure using a Terraform template via Port Actions.
+
+The workflow is executed through a Jenkins pipeline.
+
+## Prerequisites
+1. Install the following plugins in Jenkins: 
+   1. Azure Credentials - This plugin provides the `Azure Service Principal` kind in Jenkins Credentials.
+   2. [Terraform Plugin](https://plugins.jenkins.io/terraform/)
+
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:15]: Add as a prerequisite a GitHub repository to contain the Terraform template files

Comment (MPTG94) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:15]: You are also using the webhook plugin for Jenkins, so it also needs to be mentioned somewhere in the doc, see the example from here:
image

Comment (MPTG94) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:15]: This is taken from the Scaffold bitbucket repositories using cookiecutter doc

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 10]
```diff
@@ -0,0 +1,460 @@
...
+            "endpoint"         = azurerm_storage_account.storage_account.primary_web_endpoint
+            }
+        }
+
+        depends_on = [azurerm_storage_account.storage_account]
+    }
+  ```
+</details>
+
+:::tip
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:195]: This tip block is unnecessary, please remove

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Please make changes according to the comments

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 11]
```diff
@@ -0,0 +1,460 @@
...
+
+# Create resource in Azure Cloud with Terraform
+
+This example demonstrates how to deploy a storage account in Azure using a Terraform template via Port Actions.
+
+The workflow is executed through a Jenkins pipeline.
+
+## Prerequisites
+1. Install the following plugins in Jenkins: 
+   1. Azure Credentials - This plugin provides the `Azure Service Principal` kind in Jenkins Credentials.
```
[the remark this answers is not in the document]
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:13]: added

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 1 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:27]: added

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 2 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:35]: added

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 3 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:84]: modified

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 4 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:87]: Oh, naming it `JOB_TOKEN` is better. modified this too

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 6 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:328]: I have revised this now

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 7 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:134]: revised this

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 8 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:135]: That's correct. It helps to outline what the files are for. I will add this.

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 9 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:15]: Alright, I have done this.

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 10 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:195]: I have removed it now.

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 12]
```diff
@@ -0,0 +1,443 @@
...
+   2. [Terraform Plugin](https://plugins.jenkins.io/terraform/)
+   3. [Generic Webhook Trigger](https://plugins.jenkins.io/generic-webhook-trigger/)
+
+## Example - creating a storage account
+
+Follow these steps to get started:
+
+1. Create the following as Jenkins Credentials:
+    1. Create the Port Credentials using the `Username with password` kind and the id `port-credentials`.
+        1. `PORT_CLIENT_ID` - Port Client ID [learn more](../../../../build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:26]: These URLs do not work, please use absolute paths (start from `/build-your-software-catalog`) and make sure the URLs work

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 13]
```diff
@@ -0,0 +1,443 @@
...
+# Create resource in Azure Cloud with Terraform
+
+This example demonstrates how to deploy a storage account in Azure using Terraform templates via Port Actions.
+
+The workflow is executed through a Jenkins pipeline.
+
+## Prerequisites
+1. Install the following plugins in Jenkins: 
+   1. [Azure Credentials](https://plugins.jenkins.io/azure-credentials/) - This plugin provides the `Azure Service Principal` kind in Jenkins Credentials.
+   2. [Terraform Plugin](https://plugins.jenkins.io/terraform/)
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:17]: Please add a short description for these two plugins as well

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 14]
```diff
@@ -0,0 +1,443 @@
...
+## Example - creating a storage account
+
+Follow these steps to get started:
+
+1. Create the following as Jenkins Credentials:
+    1. Create the Port Credentials using the `Username with password` kind and the id `port-credentials`.
+        1. `PORT_CLIENT_ID` - Port Client ID [learn more](../../../../build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
+        2. `PORT_CLIENT_SECRET` - Port Client Secret [learn more](../../../../build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
+    2. Create the Azure Credentials using the `Azure Service Principal` kind and the id `azure`.
+        :::info NOTE
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:29]:
```suggestion
        :::tip
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 15]
```diff
@@ -0,0 +1,443 @@
...
+    "mirrorProperties": {},
+    "calculationProperties": {},
+    "relations": {}
+}
+```
+
+  </details>
+
+
+3. Create a Port Action in the [self-service hub](https://app.getport.io/self-serve) using the following JSON definition:
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:88]:
```suggestion
3. Create a Port action in the [self-service hub](https://app.getport.io/self-serve) using the following JSON definition:
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 5 above)
Comment (MPTG94) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:101]: @phalbert please apply the fix to the icon

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 16]
```diff
@@ -0,0 +1,443 @@
...
+    "trigger": "CREATE",
+    "requiredApproval": false
+}
+```
+
+  </details>
+
+4. Create the following terraform templates in a `terraform` folder in the root of your GitHub repository:
+    1. `main.tf` - This file will contain the resource blocks which defines the Storage Account to be created in the Azure cloud and the entity to be createed in Port.
+    2. `variables.tf` – This file will contain the variable declarations that will be used in the resource blocks e.g. the port credentials and port run id.
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:140]:
```suggestion
    2. `variables.tf` – This file will contain the variable declarations that will be used in the resource blocks e.g. the Port credentials and Port run id.
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 17]
```diff
@@ -0,0 +1,443 @@
...
+    3. `output.tf` – This file will contain the url of the Storage Account that needs to be generated on successful completion of “apply” operation. This url will be used in the `endpoint` property when creating the Port entity.
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:141]:
```suggestion
    3. `output.tf` – This file will contain the URL of the Storage Account that needs to be generated on successful completion of an “apply” operation. This URL will be used in the `endpoint` property when creating the Port entity.
```

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 18]
```diff
@@ -0,0 +1,443 @@
...
+
+<details>
+  <summary>Terraform `main.tf` template</summary>
+  
+  ```yaml
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:146]: Please make sure **all** codeblocks have `showLineNumbers` in them

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 19]
```diff
@@ -0,0 +1,443 @@
...
+        }
+
+        depends_on = [azurerm_storage_account.storage_account]
+    }
+  ```
+</details>
+
+
+<details>
+  :::note
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:204]: Move the `:::note` block underneath the `<summary>` tag, it doesn't change to how the markdown renders but it makes it easier to edit the file in the future

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 20]
```diff
@@ -0,0 +1,443 @@
...
+                    patterns: [[pattern: '.gitignore', type: 'INCLUDE'],
+                               [pattern: '.propsfile', type: 'EXCLUDE']])
+        }
+    }
+}
+```
+
+</details>
+
+6. Trigger the action from the [Self-service](https://app.getport.io/self-serve) tab of your Port application.
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:443]:
```suggestion
6. Trigger the action from the [self-service](https://app.getport.io/self-serve) tab of your Port application.
```

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Please see my additional comments

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 12 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:26]: Let me make certain all URLs work.

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 13 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:17]: I have added these descriptions.

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 18 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:146]: Alright then.

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 19 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:204]: I have moved it.

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md [hunk 21]
```diff
@@ -0,0 +1,441 @@
...
+   2. [Terraform](https://plugins.jenkins.io/terraform/) - This plugin provides a build wrapper that simplifies the execution of Terraform commands, such as apply, plan, and destroy.
+   3. [Generic Webhook Trigger](https://plugins.jenkins.io/generic-webhook-trigger/) - This plugin enables Jenkins to receive and trigger jobs based on incoming HTTP requests, extracting data from JSON or XML payloads and making it available as variables.
+
+## Example - creating a storage account
+
+Follow these steps to get started:
+
+1. Create the following as Jenkins Credentials:
+    1. Create the Port Credentials using the `Username with password` kind and the id `port-credentials`.
+        1. `PORT_CLIENT_ID` - Port Client ID [learn more](https://docs.getport.io/build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:26]: While the links work now, you should utilize the internal linking of the docs system.

Please change the link value to: `/build-your-software-catalog/sync-data-to-catalog/api/#get-api-token`

After that, verify that it works. Try to make the same use of absolute paths to other files in the docs project in the future.

The internal paths allow the linking to provide a better experience for people reading the docs, and using absolute paths (starting with `/` ) in the links makes it easier to move pages around

**Review by MPTG94 — CHANGES_REQUESTED** [review]
One last comment

File: docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md (hunk 21 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/jenkins-pipeline/examples/create-azure-resource.md:26]: Thank you for this explanation. It make clear the reason for internal linking.

**Review by MPTG94 — APPROVED** [review]
LGTM

**Pull request merged**